### PR TITLE
Fix: return allowed transaction isolation level value on select query

### PR DESCRIPTION
**Pull request by harshit-gangal**
## Description

This PR fixes the regression caused by https://github.com/vitessio/vitess/pull/11673 where when transaction isolation is not set returns `DEFAULT` which is an unknown value from the transaction isolation system variable.
Now, this PR will return the default value from the connection which will be one of these
`READ-UNCOMMITTED`, `READ-COMMITTED`, `REPEATABLE-READ` and `SERIALIZABLE`

## Related Issue(s)

- Fixes https://github.com/vitessio/vitess/issues/11793

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [ ] Tests were added or are not required
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/vt/sqlparser/ast_rewriting_test.go [hunk 1]
```diff
@@ -442,6 +426,27 @@ func TestRewritesSysVar(in *testing.T) {
...
+	}, {
+		in:       "SELECT @@transaction_isolation",
+		expected: "select @@transaction_isolation from dual",
+	}, {
+		in:       "SELECT @@session.transaction_isolation",
+		expected: "select @@session.transaction_isolation from dual",
+	}, {
+		in:       "SELECT @@tx_isolation",
+		expected: "select :__vttx_isolation as `@@tx_isolation` from dual",
+		sysVar:   map[string]string{"tx_isolation": "'READ-COMMITTED'"},
```
Comment (systay) [inline go/vt/sqlparser/ast_rewriting_test.go:441]: nit: I found it difficult to read these cases, until I realised that sysVar is an input and not an output. `expected` should come last

File: go/vt/vtgate/engine/set.go [hunk 2]
```diff
@@ -472,11 +472,14 @@ func (svss *SysVarSetAware) Execute(ctx context.Context, vcursor VCursor, env *e
 		if err != nil {
 			return err
 		}
-		out, ok := isolationLevelToExecuteOptionIsolationLevel[strings.ToLower(str)]
+		upperStr := strings.ToUpper(str)
```
Comment (systay) [inline go/vt/vtgate/engine/set.go:475]: curious to why you switched from lower to upper. not important, just curious :)

File: go/vt/vtgate/engine/set.go [hunk 3]
```diff
@@ -472,11 +472,14 @@ func (svss *SysVarSetAware) Execute(ctx context.Context, vcursor VCursor, env *e
...
+		out, ok := isolationLevelToExecuteOptionIsolationLevel[upperStr]
 		if !ok {
 			return vterrors.NewErrorf(vtrpcpb.Code_INVALID_ARGUMENT, vterrors.WrongValueForVar, "Variable '%s' can't be set to the value of '%s'", svss.Name, str)
 		}
-		vcursor.Session().SetTransactionIsolation(querypb.ExecuteOptions_TransactionIsolation(out))
+		vcursor.Session().SetTransactionIsolation(out)
+		vcursor.Session().SetSysVar(sysvars.TxIsolation.Name, fmt.Sprintf("'%s'", upperStr))
```
Comment (systay) [inline go/vt/vtgate/engine/set.go:481]: I'm not really understanding why you write it twice to the session under different names. is it really needed?

File: go/vt/vtgate/engine/set.go (hunk 2 above)
Comment (harshit-gangal) [reply go/vt/vtgate/engine/set.go:475]: mysql returns in upper case, so how we store, we return that after rewrite.

File: go/vt/vtgate/engine/set.go (hunk 3 above)
Comment (harshit-gangal) [reply go/vt/vtgate/engine/set.go:481]: tx_isolation is the old system variable, transaction_isolation was introduced in 5.7.20. 
In 8.0, tx_isolation is removed.
just to support the case where the client set tx_isolation and selects transaction_isolation, this is done.

Comment (systay) [reply go/vt/vtgate/engine/set.go:481]: ah, good to know. maybe we should add a comment saying this?

**Review by systay — APPROVED** [review]

**Review by frouioui — APPROVED** [review]
I am not very knowledgable on transaction isolation, but code bits that I understand look good to me.

**Pull request merged**